FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -49,7 +50,28 @@ public enum ModelDownloadType {
 }
 
 /// Downloader to manage custom model downloads.
-public struct ModelDownloader {
+public class ModelDownloader {
+  /// FirebaseApp associated with this instance of ModelDownloader.
+  private let app: FirebaseApp
+
+  private init(app: FirebaseApp) {
+    self.app = app
+  }
+
+  /// Model downloader with default app.
+  static func modelDownloader() throws -> ModelDownloader {
+    guard let defaultApp = FirebaseApp.app() else {
+      // TODO: Replace with more appropriate error.
+      throw DownloadError.internalError(description: "Default Firebase app not configured.")
+    }
+    return modelDownloader(app: defaultApp)
+  }
+
+  /// Model Downloader with custom app.
+  static func modelDownloader(app: FirebaseApp) -> ModelDownloader {
+    return ModelDownloader(app: app)
```

Comment:
Right now this instantiates a new instance every team. We need to keep a static dictionary of all the configured instances keyed on FirebaseApp name and return that instance if requested again. The components system does that for us, but since we're not using it yet well have to manually do so.

Reply:
Gotcha, done.